modules/apps/portal/portal-tika/src/main/java/com/liferay/portal/tika/internal/util/TikaConfigHelper.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.tika.internal.util;
+
+import com.liferay.petra.io.StreamUtil;
+import com.liferay.petra.io.unsync.UnsyncByteArrayInputStream;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.exception.SystemException;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.tika.internal.configuration.TikaConfiguration;
+
+import java.io.IOException;
+
+import java.util.Map;
+
+import org.apache.tika.config.TikaConfig;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.parser.Parser;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * @author Shuyang Zhou
+ * @author Jorge Díaz
+ */
+@Component(
+	configurationPid = "com.liferay.portal.tika.internal.configuration.TikaConfiguration",
+	service = TikaConfigHelper.class
+)
+public class TikaConfigHelper {
+
+	public String[] getTextExtractionForkProcessMimeTypes() {
+		return _tikaConfiguration.textExtractionForkProcessMimeTypes();
+	}
+
+	public TikaConfig getTikaConfig() {
+		return _tikaConfig;
+	}
+
+	public String getTikaConfigXml() throws IOException {
+		String tikaConfigXml = _tikaConfiguration.tikaConfigXml();
```

Comment:
@jorgediaz-lr Base on the logic here, the value of the property tikaConfigXml would be the xml content rather that the xml path. This may cause other issues when reading xml content as a property value. Let's pass in xml path instead, and at runtime use current bundle class loader load that file first, if failed, fall back to use portal class loader to load the file.

Customer can either put the file in portal class path or use a fragment bundle.

Tina.